scanner/indexer/indexer.go

```diff
@@ -182,15 +226,65 @@ func NewIndexer(ctx context.Context, cfg config.IndexerConfig) (Indexer, error)
 		return nil, err
 	}
 
+	// Use indexer.Ecosystems instead of the ecosystems we pass to libindex.New
+	// in case libindex.New adds any ecosystems (which it does).
+	vscnrs, err := versionedScanners(ctx, indexer.Ecosystems)
+	if err != nil {
+		return nil, err
+	}
+
+	manifestManager := manifest.NewManager(ctx, metadataStore, locker)
+	// Set any manifests indexed prior to the existence of the manifest_metadata table
+	// to expire immediately.
+	err = manifestManager.MigrateManifests(ctx, time.Now())
+	if err != nil {
+		return nil, fmt.Errorf("migrating manifests to metadata store: %w", err)
+	}
```

Comment:
`MigrateManifests` uses a `TryLock`, so it won't block other replicas. The query may potentially look at a bunch of rows, but that may be ok.

```
WHERE NOT EXISTS (
	SELECT FROM manifest_metadata mm WHERE mm.manifest_id = m.hash
)
```

I'm not a DB expert, but I imagine the slowest part of the query would be this part, which I think may be decently fast, as we are comparing primary keys, which (I believe) are indexed, by default.

So, I think this is ok to do upon startup. Also, from what I can tell, there are no timeouts (I guess this is both good and bad since we don't have to worry about it taking a while, but we have nothing stopping it from halting forever).

I'm ok with keeping this here for now and creating a ticket to consider moving this outside the initialization step

Reply:
Linked a Jira ticket: [032edec](https://github.com/stackrox/stackrox/pull/12648/commits/032edec4b65bcc8d1852bd3b4ba09aaa5feb940d)